mlir/lib/Dialect/MemRef/IR/MemRefOps.cpp

```diff
@@ -1615,8 +1615,10 @@ GetGlobalOp::verifySymbolUses(SymbolTableCollection &symbolTable) {
 //===----------------------------------------------------------------------===//
 
 LogicalResult LoadOp::verify() {
-  if (getNumOperands() != 1 + getMemRefType().getRank())
-    return emitOpError("incorrect number of indices for load");
+  if (getNumOperands() - 1 != getMemRefType().getRank()) {
+    return emitOpError("incorrect number of indices for load, expected ")
+           << getMemRefType().getRank() << " but got " << getNumOperands() - 1;
+  }
```

Comment:
I moved the `+ 1` to the other side because, for example, `memref.load %1[%arg4]` has `%1` as its first operand and `%arg4` as it's second operand. So the number of indices for `memref.load` is `getNumOperands() - 1`.

Reply:
I may be missing something obvious :sweat_smile:, but is there a reason why this is not just:
 ```suggestion
  if (getIndices().size() != getMemRefType().getRank()) {
    return emitOpError("incorrect number of indices for load, expected ")
           << getMemRefType().getRank() << " but got " <<getIndices().size();
  }
```

Reply:
Thanks 👍 Nice suggestion!